### PR TITLE
correct appveyor build

### PR DESCRIPTION
**Pull request by chcg**
- correct appveyor build by using new PCRE download site
- prepare for VS2019\VS2022

Questions:
- What about using newer VS version?
- What about using newer PCRE2 version. Current one is 10.40.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -16,7 +16,7 @@ param(
     [string] $arch = "x64",
     
 
-    [ValidateSet("15","14","12")]
+    [ValidateSet("17","16","15","14","12")]
```
Comment (xuhdev) [inline build.ps1:19]: Could you drop version 12 here?

**Review by xuhdev — COMMENTED** [review]
Thank you for your contribution! Yes, later versions of VS and pcre are welcome. We can drop the support of older versions.

**Comment by chcg** [comment]
Use VS2022 and 10.37 for current builds. With 10.38 there seems to be a change, so PCRE2 libs are no longer found by cmake because the lib name then contains static within the name.

**Comment by ffes** [comment]
I would suggest switching to GitHub Actions for all CI. It works well to run Visual Studio (I know from experience). For Linux there is no real reason to keep using AppVeyor for that. But I am not a maintainer of this repo, so it is not my decision.

**Review by xuhdev — APPROVED** [review]

**Comment by xuhdev** [comment]
Thank you for your contribution! All sounds good.

Since we are dropping Travis CI, I'm fine with either AppVeyor or GitHub CI. PRs are welcome to make the switch!

File: build.ps1 (hunk 1 above)
Comment (3da4ya) [reply build.ps1]: param(
    [string] $arch = "x64"
)

# ValidateSet attribute limits the possible values of $arch parameter.
[ValidateSet("17", "16", "15", "14", "12")]
[string] $arch = "x64"

# Example: Preparing environment for building
Write-Output "Preparing environment for building architecture $arch..."

# Function to initialize the build process
function Initialize-Build {
    param (
        [string] $architecture
    )
    Write-Output "Initializing build for architecture $architecture..."
    # Add your initialization code here
}

# Function to build the project
function Build-Project {
    param (
        [string] $architecture
    )
    Write-Output "Building project for architecture $architecture..."
    # Add your build commands here, for example:
    & msbuild.exe /p:Configuration=Release /p:Platform=$architecture
}

# Function to perform post-build actions
function PostBuild-Actions {
    Write-Output "Performing post-build actions..."
    # Add your post-build commands here
}

# Main script execution
Initialize-Build -architecture $arch
Build-Project -architecture $arch
PostBuild-Actions

Write-Output "Build process completed for architecture $arch."

**Pull request merged**